src/cli/cmd/mod.rs

```diff
@@ -367,6 +368,20 @@ macro_rules! flakehub_url {
     }};
 }
 
+async fn nix_command(args: &[&str]) -> Result<(), FhError> {
+    if !command_exists("nix") {
+        return Err(FhError::MissingExecutable(String::from("nix")));
+    }
```

Comment:
Why not make `command_exists` return this error itself? Then it becomes something like `check_command_exists("bin")?;`